### DISCUSSION TITLE
binary sensors adjustment integration

### DISCUSSION THREAD

**Discussion by CorHofstee** [mbuchber/ha_heliotherm#27] [Show and tell]
managed to read alle data from my heliotherm soil heat exchanger into home assistant through the Heliotherm integration. I did this trough the local RCG Webcontrol AT webpage. 
I now try to adjust the energy use of the heatpump in time by using the EVU sperre en de Wasser Vorrang (both binary sensors). However when I try to make automations , the sensors appear to be unavailable. Is that because I  did not setup a descent Modbus connection ?. In other words how can I change the status of the two binary sensors ?

Cor Hofstee

**Comment by helyxx** [comment]
how can you read the RCG data? Would love to write on the RCG registers as passive cooling is not working smoothly and I have to adjust pump speed manul every few weeks (depending on brine temp).

**Comment by CorHofstee** [comment]
I use the Heliiotherm integration right now and put the heatpump from
Automatik to Sommer between 6 and 9 in the morning and evening (we have
dynamic prices for the electricity (Netherlands))
I would love to use the SGR parts (evu sperre (expensive prices), and put
he compressor at the maximum (low or negative prices to heat up the 500 L
boiler), but this is so far not possible.
I look around for solutions (hardware or software) and I am very lucky to
be able to read German. I know of Homey Pro but that costs 399 euro's.
Currently my battery company (victron) is also looking in the use of the
one-way SGR protocol to optimize the energy usage of heatpumps with their
Victron dynamic  ESS.
KNX may also be a solution. Mobus is more complicated as the Hass
integration use already Modbus ICT.

regards, Cor
---------- Forwarded message ---------
From: helyxx ***@***.***>
Date: Sat, 26 Oct 2024, 22:02
Subject: Re: [mbuchber/ha_heliotherm] binary sensors adjustment integration
(Discussion #27)
To: mbuchber/ha_heliotherm ***@***.***>
Cc: CorHofstee ***@***.***>, Author ***@***.***>


how can you read the RCG data? Would love to write on the RCG registers as
passive cooling is not working smoothly and I have to adjust pump speed
manul every few weeks (depending on brine temp).

—
Reply to this email directly, view it on GitHub
<https://github.com/mbuchber/ha_heliotherm/discussions/27#discussioncomment-11062587>,
or unsubscribe
<https://github.com/notifications/unsubscribe-auth/AQCAHOX4JFEDJR2EHB4CJT3Z5PYM3AVCNFSM6AAAAABQVC3ME2VHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTCMBWGI2TQNY>
.
You are receiving this because you authored the thread.Message ID:
***@***.***>

**Comment by helyxx** [comment]
There are modbus registers available via Ethernet/TCP that you can use to set your heatpump a certain power consumption level (more intelligent than the smart-grid switch).
I have posted the page with the modbus registers here
https://github.com/mbuchber/ha_heliotherm/discussions/34
I am just not capable of programming here but have a logic unit in php that works with everything they state in the manual.

**Comment by CorHofstee** [comment]
I can read the evu sperre and many other parameters through the integration
(modbus ict) but i cannot adjust many of these parameters (including RVU
sperre). Can you adjust everything with knx ?
Ik could lay another cable once I am sure it works.

Regards Cor Hofstee

**Comment by helyxx** [comment]
I can change/write all parameters in this list:
![image](https://github.com/user-attachments/assets/d51dfdfe-18d2-4696-9f72-653824a7f078)
